### ISSUE TITLE
Feature Engineering

### ISSUE THREAD

**Issue by yufang67** [yufang67/machine-learning-articles#17]
# TL;DR

### Categorical embedding
- neural net to build embedding

### Expansion encoding
- is_mobile ?
- is_latest_version ?

### Closeness to major events
- date_3_days_before_holidays
### Spatial Varalbles
- GPS-coordinates

### Categorizing location
- k-means clustering
- raw latitude longitude
- zip code

### Closeness to hubs
- closeness to major hubs

### Spatial fraudulent behavior
- impossible travel speed
- spending in a different town than home
- never spending at the same location

### Data Exploration
- outliers, noise, feature engineering, cleaning 
- correlation

### Label engineering
- Log-transform
- Box-Cox
- Create a score: binary to regression
- train regressor to predict not available features

# Ref
[feature engineering](https://mp.weixin.qq.com/s?__biz=MzIyNjM2MzQyNg==&mid=2247534028&idx=1&sn=5a1eeb20007d0a41850dbb3491eed00d&chksm=e8739481df041d9734a79bbfcd92b00a0a91a889002c75a93d7cbeeeda592d7eeb1958f9a74d&mpshare=1&scene=24&srcid=0909LB4kIo256uWmz706qB5C&sharer_sharetime=1599662763504&sharer_shareid=50db2e6276d3a235822e392fb4f1a098#rd)
